tap_linkedin_ads/client.py

```diff
@@ -228,23 +246,28 @@ def refresh_access_token(self):
 
         data = response.json()
         self.__access_token = data['access_token']
-
         # data['expires_in'] is an integer of seconds until the access_token expires.
         # Technically this self.__expires is inaccurate because it was true when LinkedIn generated the token, but
         # we receive and process that response some (very) small amount of time after it was true.
         self.__expires = datetime.utcnow() + timedelta(seconds=data['expires_in'])
 
+        self.write_access_token_to_config()
 
     def fetch_and_set_access_token(self):
+        """
+        This method generates a new access token if the refresh token is provided.
 
-        # If refresh token is not provided then we are assuming that it is an old connection
+        Note: Linkedin-ads access token expires in 60 days, whereas the refresh token expires in 365 days.
+        """
+        # If the refresh token is not provided then we are assuming that it is an old connection
         # and client has provided the valid access_token already
+        # Checking if the token is expired, It will be refreshed
         if not self.__refresh_token:
             return
 
         if self.__access_token:
-
-            if self.get_token_expires() > datetime.utcnow():
+            # Subtracting 1 day from the expiration date to avoid the failure of the token in case of a longer sync run.
```

Comment:
fetch_and_set_access_token is called upon every request that the tap makes, so a longer sync run shouldn't be a risk we're worried with. We've directly observed issues retrieving an access token too early, so unless we can find documentation from linkedin stating when we can retrieve a new access token safely, I think we should change this to something substantially lower so that it's averting the risk of a token expiring before we can complete a single request, eg: 

get_token_expires() - timedelta(seconds=10)

Reply:
Agree @dsprayberry but we have already changed that logic to call `fetch_and_set_access_token` in each request. Now, we are calling `fetch_and_set_access_token` just 1 time while making instance of client. That's why we are subtracting 1 day from the expiration time to avoid the failure of the token in case of a longer sync run.